# HG changeset patch
# User Matthew Butterworth <me@mattbutterworth.co.uk>
# Date 1792697926 -7200
# Node ID 01232de50e6841bf234e5d4f7288bcb8e4c93061
# Parent  aa48e1add824b4c8618ee68ffe98e7df8d11f36b
Bug 1805878 - Return Reply-To from nsMsgFileHdr for opened .eml files. r=mkmelin

nsMsgFileHdr never stored Reply-To, so replying to a message opened from a file fell back to the From address and could show the no-reply prompt even when a valid Reply-To was present.

Differential Revision: https://phabricator.services.mozilla.com/D335414

diff --git a/mailnews/local/src/nsMsgFileHdr.cpp b/mailnews/local/src/nsMsgFileHdr.cpp
--- a/mailnews/local/src/nsMsgFileHdr.cpp
+++ b/mailnews/local/src/nsMsgFileHdr.cpp
@@ -83,8 +83,11 @@ nsresult nsMsgFileHdr::ReadFile() {
     }
     if (name.EqualsLiteral("Bcc") && mBccList.IsEmpty()) {
       mBccList = hdr.Value(buffer);
     }
+    if (name.LowerCaseEqualsLiteral("reply-to") && mReplyTo.IsEmpty()) {
+      mReplyTo = hdr.Value(buffer);
+    }
     if (name.EqualsLiteral("Date") && mDate == 0) {
       PR_ParseTimeString(hdr.Value(buffer).get(), false, &mDate);
     }
     if (name.EqualsLiteral("Message-ID") && mMessageID.IsEmpty()) {
@@ -130,8 +133,14 @@ NS_IMETHODIMP nsMsgFileHdr::GetStringPro
   if (!strcmp(propertyName, "dummyMsgUrl")) {
     _retval = mUri;
     return NS_OK;
   }
+  if (!strcmp(propertyName, "replyTo")) {
+    nsresult rv = ReadFile();
+    NS_ENSURE_SUCCESS(rv, rv);
+    _retval = mReplyTo;
+    return NS_OK;
+  }
   _retval.Truncate();
   return NS_OK;
 }
 
diff --git a/mailnews/local/src/nsMsgFileHdr.h b/mailnews/local/src/nsMsgFileHdr.h
--- a/mailnews/local/src/nsMsgFileHdr.h
+++ b/mailnews/local/src/nsMsgFileHdr.h
@@ -37,8 +37,9 @@ class nsMsgFileHdr : public nsIMsgDBHdr 
   nsCString mRecipients;
   nsString mDecodedRecipients;
   nsCString mCcList;
   nsCString mBccList;
+  nsCString mReplyTo;
   PRTime mDate;
   nsCString mMessageID;
   uint32_t mFlags;
 
